spaceship/lib/spaceship/client.rb

```diff
@@ -484,11 +484,20 @@ def send_shared_login_request(user, password)
       # Now we know if the login is successful or if we need to do 2 factor
 
       case response.status
-      when 403
-        raise InvalidUserCredentialsError.new, "Invalid username and password combination. Used '#{user}' as the username."
       when 200
         fetch_olympus_session
         return response
+      when 401
+        # Unauthorized
+        raise InvalidUserCredentialsError.new, "Invalid username and password combination. Used '#{user}' as the username."
+      when 403
+        # Forbidden
+        if response.body["serviceErrors"][0]["code"] == "-20209"
```

Comment:
I like these smarter errors 💪

Better errors are something that make a great product and better user experience, so thanks for the PR!

A few suggestions

- For 403 we should use `UnauthorizedAccessError` instead of `InvalidUserCredentialsError`

- I think we should switch on the code since we know a few others from https://github.com/fastlane/fastlane/issues/14398

- If we really don't know what's wrong, we should provide the iForgot url and also the url to contact dev relations

for example (this is pseudo-ruby, I didn't try running it or check for typos)

```ruby
error_string = "Login worked, but access was forbidden. Try logging in on developer.apple.com and see if any issues are raised to you there."

case response.body["serviceErrors"][0]["code"]
when "-20209"
  error_string = "This Apple ID has been locked for security reasons. Visit iForgot to reset your account (https://iforgot.apple.com)."
when "-20751"
  error_string = "Login worked, but something went wrong anyway. This has happened when an Apple ID got lost on Apple's end. See https://github.com/fastlane/fastlane/issues/14398; try resetting your account on https://iforgot.apple.com; or contact developer relations https://developer.apple.com/contact/"
end

raise UnauthorizedAccessError.new, error_string

```